app/models/delay_henka/scheduled_change.rb

```diff
@@ -18,10 +18,10 @@ class ScheduledChange < ApplicationRecord
 
     def self.schedule(record:, changes:, by_id:)
       changes.each do |attribute, new_val|
-        old_val = record.public_send(attribute)
-        cleaned_new_val = cleanup_val(record, attribute, new_val)
-        next if old_val == cleaned_new_val
-        create!(changeable: record, submitted_by_id: by_id, attribute_name: attribute, old_value: old_val, new_value: cleaned_new_val)
+        cleaned_new_val = cleanup_val(new_val)
+        record.public_send("#{attribute}=", cleaned_new_val)
+        next unless record.public_send("#{attribute}_changed?")
```

Comment:
Instead of us explicitly running type-cast, this approach will invoke any overwritten setters and thus i feel this is a better solution - though i'm curious whether you have different opinion about this. feel free to let me know

Reply:
@zinosama agree with it, it's more precise,